src/platform/unix.rs

```diff
@@ -470,62 +470,52 @@ impl MonitorHandleExtUnix for MonitorHandle {
 }
 
 /// A theme for a Wayland's client side decorations.
-///
-/// The colors are in ARGB format.
+#[cfg(feature = "wayland")]
 pub trait Theme: Send + 'static {
-    /// Primary color of the scheme.
-    fn primary_color(&self, window_active: bool) -> [u8; 4];
+    /// Title bar color.
+    fn titlebar_element_color(&self, element: TitleBarElement, window_active: bool) -> ARGBColor;
 
-    /// Secondary color of the scheme.
-    fn secondary_color(&self, window_active: bool) -> [u8; 4];
-
-    /// Color for the close button.
-    fn close_button_color(&self, status: ButtonState) -> [u8; 4];
-
-    /// Icon color for the close button, defaults to the secondary color.
-    #[allow(unused_variables)]
-    fn close_button_icon_color(&self, status: ButtonState) -> [u8; 4] {
-        self.title_color(true)
-    }
-
-    /// Background color for the maximize button.
-    fn maximize_button_color(&self, status: ButtonState) -> [u8; 4];
-
-    /// Icon color for the maximize button, defaults to the secondary color.
-    #[allow(unused_variables)]
-    fn maximize_button_icon_color(&self, status: ButtonState) -> [u8; 4] {
-        self.title_color(true)
-    }
-
-    /// Background color for the minimize button.
-    fn minimize_button_color(&self, status: ButtonState) -> [u8; 4];
-
-    /// Icon color for the minimize button, defaults to the secondary color.
-    #[allow(unused_variables)]
-    fn minimize_button_icon_color(&self, status: ButtonState) -> [u8; 4] {
-        self.title_color(true)
-    }
+    /// Color for a given button part.
+    ///
+    /// When `is_icon` is `true` the returned color is applied to a button's icon.
+    /// When `is_icon` is `false` the returned color is applied to a button's background.
+    fn button_color(
+        &self,
+        button: Button,
+        is_icon: bool,
+        window_active: bool,
+        state: ButtonState,
```

Comment:
I think the event order could be refined a bit, with button/state first and icon/window_active at the end.

I also think that `is_icon` is a pretty bad parameter name. Something like `foreground` might be better? But I don't really like that either.

Could of course also return two ARGBColors, but that might make things more difficult downstream when you want to use the same background for all states or something.